crypto/evp/evp_enc.c

```diff
@@ -365,8 +367,11 @@ static int evp_cipher_init_internal(EVP_CIPHER_CTX *ctx,
             ctx->num = 0;
             /* Don't reuse IV for CTR mode */
             if (iv != NULL) {
-                if ((n = EVP_CIPHER_CTX_get_iv_length(ctx)) <= 0)
+                n = EVP_CIPHER_CTX_get_iv_length(ctx);
```

Comment:
Strange asymmetry: here a zero IV length is not allowed (and wasn't previously)?